coderd/database/queries/workspacebuildparameters.sql

```diff
@@ -14,3 +14,28 @@ FROM
     workspace_build_parameters
 WHERE
     workspace_build_id = $1;
+
+-- name: GetUserWorkspaceBuildParameters :many
+SELECT
+    sub.name,
+    sub.value,
+    sub.created_at
+FROM (
+    SELECT
+        wbp.name,
+        wbp.value,
+        wb.created_at,
+        ROW_NUMBER() OVER (PARTITION BY wbp.name ORDER BY wb.created_at DESC) as rn
+    FROM
+        workspace_build_parameters wbp
+    JOIN 
+        workspace_builds wb ON wb.id = wbp.workspace_build_id
+    JOIN
+        workspaces w ON w.id = wb.workspace_id
+    WHERE
+        w.owner_id = $1
```

Comment:
I'm sure that users will use the same parameter names but with different meanings. For example, a favorite `machine_type` for an ML template will be different than one for a standard dev workspace.

Can we require a `template_id here for better filtering of parameter values?

Reply:
The decision to use all templates was intentional as the parameters I had in mind were of the type “aws_region” and “dotfiles_uri”. If people feel strongly that it should be namespaced to template_id I can change it.

Bear in mind this decision is a two day door.

Reply:
I have a feeling that we may pull too much junk from the database, and having a parameter for "template_id" would be a first step towards a [prompt API](https://github.com/coder/coder/issues/7099). Technically, a workspace user could just receive parameters they are interested in instead of ALL workspace parameters in the database.